pinot-query-planner/src/main/java/org/apache/pinot/query/planner/physical/PinotDispatchPlanner.java

```diff
@@ -120,16 +117,6 @@ private void runValidations(PlanFragment planFragment, DispatchablePlanContext c
     }
   }
 
-  // TODO: Switch to Worker SPI to avoid multiple-places where workers are assigned.
-  private void runPhysicalOptimizers(PlanNode subPlanRoot, DispatchablePlanContext dispatchablePlanContext,
-      TableCache tableCache) {
-    if (dispatchablePlanContext.getPlannerContext().getOptions().getOrDefault("useColocatedJoin", "false")
-        .equals("true")) {
```

Comment:
for folks without context, the Greedy* code was called only if you had set the `useColocatedJoin` flag. This feature had broken around a year back due to MSE evolution, and the design of this class being outdated.

Hence removing this should be safe.